packages/firebase_ml_vision/lib/src/face_detector.dart

```diff
@@ -4,18 +4,230 @@
+  /// processing consecutive frames.
+  final bool enableTracking;
+
+  /// The smallest desired face size.
+  ///
+  /// Expressed as a proportion of the width of the head to the image width.
+  final double minFaceSize;
+
+  /// Option for controlling additional accuracy / speed trade-offs.
+  final FaceDetectorMode mode;
+}
+
+/// Represents a face detected by [FaceDetector].
+class Face {
+  Face._(dynamic data)
+      : boundingBox = Rectangle<int>(
+          data['left'],
+          data['top'],
+          data['width'],
+          data['height'],
+        ),
+        headEulerAngleY = data['headEulerAngleY'],
+        headEulerAngleZ = data['headEulerAngleZ'],
+        leftEyeOpenProbability = data['leftEyeOpenProbability'],
+        rightEyeOpenProbability = data['rightEyeOpenProbability'],
+        smilingProbability = data['smilingProbability'],
+        trackingId = data['trackingId'],
+        _landmarks = _getLandmarks(data['landmarks']);
+
+  final Map<FaceLandmarkType, FaceLandmark> _landmarks;
+
+  /// The axis-aligned bounding rectangle of the detected face.
+  final Rectangle<int> boundingBox;
+
+  /// The rotation of the face about the vertical axis of the image.
+  final double headEulerAngleY;
+
+  /// The rotation of the face about the axis pointing out of the image.
+  final double headEulerAngleZ;
+
+  /// Probability that the face's left eye is open.
+  ///
+  /// Returns a value between 0.0 and 1.0. Returns `null` if probability was not
+  /// computed.
+  final double leftEyeOpenProbability;
+
+  /// Probability that the face's right eye is open.
+  ///
+  /// Returns a value between 0.0 and 1.0. Returns `null` if probability was not
+  /// computed.
+  final double rightEyeOpenProbability;
+
+  /// Probability that the face is smiling.
+  ///
+  /// Returns a value between 0.0 and 1.0. Returns `null` if probability was not
+  /// computed.
+  final double smilingProbability;
+
+  /// The tracking ID if the tracking is enabled.
+  ///
+  /// Returns `null` if tracking was not enabled.
+  final int trackingId;
+
+  /// Gets the landmark based on the provided [FaceLandmarkType].
+  ///
+  /// Returns null if landmark was not detected.
+  FaceLandmark landmark(FaceLandmarkType landmark) => _landmarks[landmark];
+}
+
+/// Represent a face landmark.
+///
+/// A landmark is a point on a detected face, such as an eye, nose, or mouth.
+class FaceLandmark {
+  FaceLandmark._(this.type, this.position);
+
+  /// The [FaceLandmarkType] of this landmark.
+  final FaceLandmarkType type;
+
+  /// Gets a 2D point for landmark position.
+  ///
+  /// The point (0, 0) is defined as the upper-left corner of the image.
+  final Point<int> position;
+}
+
+Map<FaceLandmarkType, FaceLandmark> _getLandmarks(dynamic data) {
+  final Map<FaceLandmarkType, FaceLandmark> landmarks =
+      <FaceLandmarkType, FaceLandmark>{};
+
+  data.forEach((dynamic key, dynamic point) {
+    FaceLandmarkType landmarkType;
+    switch (key) {
```

Comment:
This `switch` could be replaced by a lookup into a `Map<String, FaceLandmarkType>`:
```dart
enum FaceLandmarkType { ... }

Map<String, FaceLandmarkType> _stringToTypeMap;

String _typeToString(FaceLandmarkType type) {
  final String full = type.toString();
  return full.substring(full.indexOf('.') + 1);
}

FaceLandmarkType _stringToType(String string) {
  _stringToTypeMap ??= Map<String, FaceLandmarkType>.fromIterables(
    FaceLandmarkType.values.map<String>(_typeToString),
    FaceLandmarkType.values,
  );
  FaceLandmarkType type = _stringToTypeMap[string];
  if (type == null) {
    throw ArgumentError.value(string, 'Landmark name', 'Not valid landmark.');
  }
  return type;
}
```

Reply:
I'm not sure how important readability is, but i also got this to work. haha

```dart
_landmarks = Map<FaceLandmarkType, FaceLandmark>.fromIterables(
          FaceLandmarkType.values,
          List<FaceLandmark>.generate(
            FaceLandmarkType.values.length,
            (index) {
              final FaceLandmarkType type = FaceLandmarkType.values[index];
              return FaceLandmark._(
                type,
                Point<int>(
                  data['landmarks'][_enumToString(type)][0],
                  data['landmarks'][_enumToString(type)][1],
                ),
              );
            },
          ),
        );
```

Reply:
It would fail if missing one though. I'll probably have to add a check.